platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java

```diff
@@ -177,17 +196,22 @@ public void update() {
      *
      * @return the average times, for each member, in milliseconds
      */
-    public double[] getPingMilliseconds() {
-        final double[] times = new double[avgPingMilliseconds.size()];
-        for (int i = 0; i < times.length; i++) {
-            times[i] = avgPingMilliseconds.get(i).get();
-        }
-        times[selfId.getIdAsInt()] = 0;
+    @NonNull
+    public Map<NodeId, Double> getPingMilliseconds() {
```

Comment:
This method could be replaced with a new `private` method, which calculates `pingValue` without creating a new map each time. Then this method could be deprecated

Up to you whether this is outside the scope of this issue